godtools/App/Features/Dashboard/Presentation/Dashboard/DashboardViewModel.swift

```diff
@@ -107,37 +93,121 @@ class ToolsMenuViewModel: ToolsMenuViewModelType {
+        self.getLessonsUseCase = getLessonsUseCase
+        self.getOptInOnboardingBannerEnabledUseCase = getOptInOnboardingBannerEnabledUseCase
+        self.getSettingsParallelLanguageUseCase = getSettingsParallelLanguageUseCase
+        self.getSettingsPrimaryLanguageUseCase = getSettingsPrimaryLanguageUseCase
+        self.getShouldShowLanguageSettingsBarButtonUseCase = getShouldShowLanguageSettingsBarButtonUseCase
+        self.getSpotlightToolsUseCase = getSpotlightToolsUseCase
+        self.getToolCategoriesUseCase = getToolCategoriesUseCase
+        self.getToolIsFavoritedUseCase = getToolIsFavoritedUseCase
+        self.removeToolFromFavoritesUseCase = removeToolFromFavoritesUseCase
+        self.toggleToolFavoritedUseCase = toggleToolFavoritedUseCase
+        
+        allToolsTabTitle = localizationServices.stringForMainBundle(key: "tool_menu_item.tools")
+        favoritesTabTitle = localizationServices.stringForMainBundle(key: "my_tools")
+        lessonsTabTitle = localizationServices.stringForMainBundle(key: "tool_menu_item.lessons")
+        selectedTab = startingTab
+        
+        tabChanged()
     }
+}
+
+// MARK: - Private
+
+extension DashboardViewModel {
     
-    func toolbarWillAppear() -> ToolsMenuToolbarViewModelType {
-        return ToolsMenuToolbarViewModel(localizationServices: localizationServices)
+    private func tabChanged() {
+        
+        let shouldShowLanguageSettingsButton = getShouldShowLanguageSettingsBarButtonUseCase.getShouldShowLanguageSettingsBarButton(for: selectedTab)
+        
+        let languageButtonCurrentlyShowing = shouldShowLanguageSettingsBarButtonItemPublisher.value.shouldShowButton
+        if languageButtonCurrentlyShowing != shouldShowLanguageSettingsButton {
+            
+            publishNewLanguageSettingsBarButtonItemState(shouldShowButton: shouldShowLanguageSettingsButton)
+        }
     }
     
-    func menuTapped() {
-        flowDelegate?.navigate(step: .menuTappedFromTools)
+    private func publishNewLanguageSettingsBarButtonItemState(shouldShowButton: Bool) {
+        
+        if shouldShowButton {
+
+            chooseLanguageButton = createLanguageBarButtonItem()
+            shouldShowLanguageSettingsBarButtonItemPublisher.send((true, chooseLanguageButton))
+            
+        } else {
+            
+            shouldShowLanguageSettingsBarButtonItemPublisher.send((false, chooseLanguageButton))
+            chooseLanguageButton = nil
+        }
     }
     
-    func languageTapped() {
+    private func createLanguageBarButtonItem() -> UIBarButtonItem {
+        let languageBarButtonItem = UIBarButtonItem()
+        languageBarButtonItem.image = ImageCatalog.navLanguage.uiImage
+        languageBarButtonItem.tintColor = .white
+        languageBarButtonItem.target = self
+        languageBarButtonItem.action = #selector(languageBarButtonItemTapped)
+        
+        return languageBarButtonItem
+    }
+}
+
+// MARK: - Public
```

Comment:
Hey @rachaelblue one other thing I would like to start doing in our ViewModels is create an extension with ```// MARK: - Inputs```.  
I'll put this in the programming guide read me.  But this way we can jump into a ViewModel and quickly get a read out of the inputs coming from the View. 
Eventually there might be a way we can have some programmatic solution on our ViewModels that we'd define inputs and outputs between the View and ViewModel, but for now marking those in an extension will do.